autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -481,14 +507,16 @@ def impute_nan_in_categories(self,
             if X[column].isna().any():
                 if column not in self.dict_missing_value_per_col:
                     try:
-                        float(X[column].dropna().values[0])
+                        first_value = X[column].dropna().values[0]
+                        float(first_value)
                         can_cast_as_number = True
                     except Exception:
                         can_cast_as_number = False
```

Comment:
we do not want to save ` first_value` as float, or did I miss something.

Reply:
I was thinking of something like the following:
```
def can_cast_as_number(value: Union[int, float, str]) -> bool:
    try:
        float(first_value)
        can_cast_as_number = True
    except ValueError:
        can_cast_as_number = False

first_value = X[column].dropna().values[0]
if can_cast_as_number(first_value):
    # process
```

Just in case, we have not done `first_value = float(first_value)`, so the type of `first_value` does not change.
```
def try_cast(value):
    try:
        float(value)
    except ValueError:
        pass


val = 1
print(type(val))
try_cast(val)
print(type(val))

val = 1.0
print(type(val))
try_cast(val)
print(type(val))

val = '1.0'
print(type(val))
try_cast(val)
print(type(val))

[Output]
<class 'int'>
<class 'int'>
<class 'float'>
<class 'float'>
<class 'str'>
<class 'str'>
```

Reply:
ah yeah yeah, I do know the value would not change, since you wrote not a method in parentheses I did not get the main idea. Sure, looks like it makes the code easier to read.